### DISCUSSION TITLE
get count of the records, even using OFFSET and LIMIT?

### DISCUSSION THREAD

**Discussion by reunix** [prisma/prisma#20241] [Q&A] [answered]
hi ,
there is a way to get the counter of the records, even using
`SKIP : 0` and `TAKE: 100` with `findMany()` ? in the same query?

```
LIMIT 100   
OFFSET 0
```

I need the totald and records to update the total amount of pages in my table in the frontend,

in this case there would only be 100 records, but the table can have 10,000, so I need the total of records as well as 100 to display on the screen, the actual filter is below:

```ts
return await this.prisma.$queryRaw`SELECT productsbase_id, productbase_nome FROM productbase 
    WHERE
    productbase_nome  LIKE  ${_whereLike} 
    ORDER BY productbase_nome
    LIMIT ${_propsFiters.size} 
    OFFSET $_propsFiters.start} ;
```

in addition to the Json, I need the total number of records, respecting only the where clause
Thanks!!

**Comment by reunix** [comment]
gpt suggested this routine,

```
const { records, total } = await prism.$transaction([
   prism.yourModel.findMany({
     skip: 40,
     take: 10,
     // Your filter criteria here
   }),
   prism.yourModel.count({
     where: {
       // Your filter criteria here
     },
   }),
]);
```

console.log(records); // retrieved records
console.log(total); // Total records

however it would be two requests in the database, is there a way to make a single request?

**Reply by nurul3101** [reply, marked as answer]
Hello @reunix 👋 

I believe it's not possible yet to achieve this in a single database request, I would recommend the same solution that GPT suggested.

I would recommend you to go through this similar GitHub Discussion:

- https://github.com/prisma/prisma/discussions/3087

We also have this Feature Request that might make it much easier to achieve in future:

- https://github.com/prisma/prisma/issues/7550

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by reunix** [reply]
Ok @nurul3101 tks!!

I will analyze the options, and wait for new features

**Reply by nurul3101** [reply]
No worries! :)
